platform/darwin/src/MGLVectorStyleLayer.h

```diff
@@ -47,7 +47,7 @@ MGL_EXPORT
  ```swift
  let layer = MGLLineStyleLayer(identifier: "contour", source: terrain)
  layer.sourceLayerIdentifier = "contours"
- layer.predicate = NSPredicate(format: "(index == 5 || index == 10) && ele >= 1500.0")
+ layer.predicate = NSPredicate(format: "(index == 5 || index == 10) && CAST(ele, 'NSNumber') >= 1500.0")
```

Comment:
This illustrates the fact that many existing uses of predicates in v3._x_ will need to be modified slightly when migrating to v4.0.0. On balance, I think this is necessary until mapbox/mapbox-gl-js#6459, but it does mean we’ll need to be clear about this requirement in the documentation, particularly the function-to-expression migration guide and the “Predicates and Expressions” guide.

Reply:
Also, the iOS and macOS changelogs should have a blurb noting that:

* Predicate format strings can contain arithmetic and calls to built-in NSExpression functions.
* Key paths may need to be cast to NSString or NSNumber in some cases.